### ISSUE TITLE
Create images specific to a given specie?

### ISSUE THREAD

**Issue by nunogit** [bridgedb/docker#6]

**Comment by marvinm2** [comment]
For personal use, it would save lots of time downloading and running the docker if you have species specific. For cloud, minishift/openshift, other online location, why not all species at once?

**Comment by Chris-Evelo** [comment]
Most people will not use more than 1 or 2 species, and certain combinations make a lot more sense than others. No one wants rice and cows for instance, unless you really want all. But people might want human + mouse + rat.

**Comment by nunogit** [comment]
Ok. That makes a lot of sense... But than how can we elect the important species to package?

**Comment by Chris-Evelo** [comment]
I am not sure, How much effort will be involved in making these containers, and how easy will it be to update them? I am tempted to make:
1) An empty one with instructions how to add.
2) A human research related one that has human, mouse, rat (and possibly also the metabolomics, reactions and gene-variant databases)
3) A complete one with everything we have.

And finally, indicate that people can contact us for other specific combinations they might want.

    Best, Chris

**Comment by nunogit** [comment]
Well, the build of a full image takes about 1 hour + upload (~20 minutes).

So, I think that idea can work. The empty one can prompt people for which databases they want. It can be quite easy.

I still have a question though. We currently have Bridgedb releases. We should probably release a docker image each time, not? For reproducibility... or is that not relevant for this?

If the answer is yes and we decide to have a container that it's custome build (1) we should also allow people to pick their own version of the bridgedb database. If that happens we should keep an history of bridgedb files?!

**Comment by Chris-Evelo** [comment]
Yes, I think both make sense. Update the containers with the releases, and archive at least the mapping files. Egon suggested to do the latter also on Zenodo or Figshare.

**Comment by nunogit** [comment]
Ok. I have to look at their storage capabilities. With FAIR I've been looking at archive.org 
Might be an interesting solution to store data like the bdb files.

**Comment by egonw** [comment]
@nunogit, can we have these Docker images be built on Jenkins?

I suggest we anticipate to have one general Docker, which downloads all files (ideally from the archive, Figshare, Archive.org, Zenodo, or otherwise; I don't have a strong preference, as long as it gives DOIs :)... @Chris-Evelo, with a FAIRMappings.org idea, that could be the source of information where to download the latest files...

And we have one or more dedicated images... @marvinm2, can you check the use cases to see which species we need for OpenRiskNet (see https://github.com/bridgedb/docker/issues/9) ?

**Comment by nunogit** [comment]
> @nunogit, can we have these Docker images be built on Jenkins?

@egonw I don't see why not. I haven't done it yet.

>  I don't have a strong preference, as long as it gives DOIs :)... 

I am no sure archive.org gives DOIs. 
In any case we should start creating a mechanism to publish the bdb databases systematically into those repos.

> which downloads all files

So, is your opinion that the databases should be downloaded after downloading the image?

**Comment by Chris-Evelo** [comment]
I am not sure... I am OK with Zenodo and Figshare as a backup facility that happens to give DOI based persistent IDs. But I am not sure that means that we should not maintain our own data repository which can also easily hold other information. I would not be surprised that we actually have to. But I understand the idea that you need a persistent dataset ID. Maybe we can work with identifiers.org and discuss in the ELIXIR identifiers workgroup whether that really should be a DOI, and if so identifiers.org can possible become an issueing instance?

**Comment by Chris-Evelo** [comment]
BTW if you indeed download the mapping files after installing BridgeDb it should not be so hard to add a download configuration file where you simply select which datasets you want.

**Comment by egonw** [comment]
Another thought is that even single species Docker images make sense... because the way OpenRiskNet works, but as I learned this week, platforms like Phenomenal too, they spin up services on demand. And people just as easily spin up two Dockers (one for each species) as one Docker (with both species).

**Comment by marvinm2** [comment]
Is this still wanted? Production settled on a single image with the data baked in (`bigcatum/bridgedb:3.0.31-2.1.9`, no volume), and per-species distribution now happens at the data layer instead — `gene.json` lists 36 species, each with its own figshare download and Zenodo DOI.

That said, smaller per-species or subset images on Docker Hub would still be useful in their own right, so this is not obviously obsolete. @egonw, worth keeping open?
